src/main/java/fr/insalyon/citi/golo/runtime/FunctionCallSupport.java

```diff
@@ -108,11 +118,15 @@ public static Object fallback(FunctionCallSite callSite, Object[] args) throws T
       Method method = (Method) result;
       checkLocalFunctionCallFromSameModuleAugmentation(method, callerClass.getName());
       types = method.getParameterTypes();
-      if (method.isVarArgs() && isLastArgumentAnArray(types.length, args)) {
+      //TODO improve varargs support on named arguments. Matching the last param type + according argument
```

Comment:
The `array[]` approach for named varargs parameter does the job. I mean everybody know that the `...` is replaced by `Object[]` so it makes sense.
I'm not sure `f(a = 1,2,3, b = 4, c = 5)` would be clearer, plus from a parsing standpoint is way harder (at least for me :smile: )

On this particular line of code, the main issue is that currently only method flagged w/ varargs + having an array as last arg are trasnformed to collect all the trailing args into an array (is the job of `asFixedArity`). Without the test I added, the varargs value is an array of arrays (a first one obtained to the `varargs` flag on the method, and a second coming from the `array[]` notation.
 
Now in that case
````
function joiner = |delim, values...| -> ...
joiner(values = array[1,2,3], delim = "-")
````
The joiner method will be flagged as varargs but the collector is in the first position not the last... 
The best option I can see  is to add metadata in the generated annotation. Basically, flag the varags collecor then during the runtime, the above `if` will check that the argument corresponding to the  flagged parameter is an instanceof `Object[]`.